doc/commitlogs.md

```diff
@@ -0,0 +1,51 @@
+## M3DB Commit Logs
+
+### Integrity Levels
+
+There are two integrity levels available for M3DB commit logs:
+
+- **Synchronous:** all write operations must synchronously finish writing an entry in the commit log to complete.
+- **Buffered:** all write operations must enqueue to a buffer an entry in the commit log to complete.
+
+Depending on the data loss requirements consumers can choose either integrity level.
+
+### Properties
+
+Commit logs will be stamped by the start time, aligned and rotated by the block size.  To restore data for an entire block you will require the commit logs from the previous block size period, the current block size period and the future block size.  This is to account for the buffer past data and the buffer future data.
```

Comment:
This will stripe writes, which is a complete anti-commit log pattern.  A good overview here http://www.benstopford.com/2015/02/14/log-structured-merge-trees/.